src/pybind/mgr/diskprediction/agent/metrics/sai_disk.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import
+
+import socket
+
+from . import MetricsAgent
+from ...common.db import DB_API
+from ...models.metrics.dp import SAI_Disk
+
+
+def get_human_readable(size, precision=2):
+    suffixes = ['B', 'KB', 'MB', 'GB', 'TB']
+    suffix_index = 0
+    while size > 1000 and suffix_index < 4:
+        # increment the index of the suffix
+        suffix_index += 1
+        # apply the division
+        size = size/1000.0
+    return "%.*d %s" % (precision, size, suffixes[suffix_index])
+
+
+class SAI_DiskAgent(MetricsAgent):
+    measurement = 'sai_disk'
+
+    @staticmethod
+    def _convert_disk_type(is_ssd, sata_version, protocol):
+        """ return type:
+            0: "Unknown", 1: "HDD",
+            2: "SSD",     3: "SSD NVME",
+            4: "SSD SAS", 5: "SSD SATA",
+            6: "HDD SAS", 7: "HDD SATA"
+        """
+        disk_type = 0
+        if is_ssd:
+            if sata_version and not protocol:
+                disk_type = 5
+            elif 'SCSI' in protocol:
+                disk_type = 4
+            elif 'NVMe' in protocol:

```

Comment:
Better check for lower case if you can't be for sure that the protocol is ALWAYS and on every platform reported as 'NVMe'. Maybe SMART (i assume it will collected by this) will report it someday as 'nvme' or 'NVME'.

Same applies to the 'SCSI' check above.